Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -545,6 +551,40 @@ class JSGlobalObject : public JSSegmentedVariableObject {
     std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>> m_numberPrototypeToStringWatchpoint;
     std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>> m_arrayBufferConstructorSpeciesWatchpoints[2];
     std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>> m_arrayBufferPrototypeConstructorWatchpoints[2];
+    std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>> m_typedArrayConstructorSpeciesWatchpoint;
+#define DECLARE_TYPED_ARRAY_TYPE_WATCHPOINT(name) \
+    std::unique_ptr<ObjectAdaptiveStructureWatchpoint> m_typedArray ## name ## ConstructorSpeciesWatchpoint; \
+    std::unique_ptr<ObjectPropertyChangeAdaptiveWatchpoint<InlineWatchpointSet>> m_typedArray ## name ## PrototypeConstructorWatchpoint;
+    FOR_EACH_TYPED_ARRAY_TYPE(DECLARE_TYPED_ARRAY_TYPE_WATCHPOINT)
+#undef DECLARE_TYPED_ARRAY_TYPE_WATCHPOINT
+
+    std::unique_ptr<ObjectAdaptiveStructureWatchpoint>& typedArrayConstructorSpeciesWatchpoint(TypedArrayType type)
```

Comment:
I know it's a bit verbose, but can we name this so it's clear it's the absence watchpoint? Same with the fields?

Reply:
This is actually checking 2 things (species absence and constructor equivalence), and only one of that is absence. If we use absence here, it is a bit misleading since this watchpoint-set is also ensuring that `constructor` property's equivalence too.
I'll rename it to `typedArrayConstructorSpeciesFastPathWatchpoint` instead.

Reply:
Ah, no, this is absence one, I'll change it to include absence.

Reply:
This one is renamed to `typedArrayConstructorSpeciesAbsenceWatchpoint`